fabtests/multinode/src/pattern/broadcast.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2017-2019 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <pattern.h>
+#include <core.h>
+
+static int pattern_next(int *cur)
+{
+	if (pm_job.my_rank == 0){
+		int next = *cur + 1;
+
+		if (next >= pm_job.num_ranks)
+			return -ENODATA;
+		else if (next == 0)

```

Comment:
nit: no need for else, since if statement above returns.
Use FI_blah error codes for consistency with rest of fabtests.

Reply:
I think this is the way that @vkrishna had had his returns with the original pattern. I think the idea is that this value tells the test runner that there is no next and to exit the send/receive loop.

Reply:
Fabtests mostly uses FI_XXX return codes.  We should be consistent and use those where possible.  There are more FI_XXX return values available than errno values.

The comment on the if-else is a nit, but refers to this coding style:
```
if (blah)
    return 0;
else
    do_other_work...
```
When the if statement returns, the 'else' is unnecessary.  This can convert to:
```
if (blah)
    return 0;

do_other_work...
```
This removes unnecessary indentation and makes it easier to read.  That benefit is lost when we're dealing with else-if, but the same concept applies.  This style comment applies to other locations in the code where else-if isn't used.

Reply:
Do you have a suggestion for which FI code to use?

The coding style stuff makes sense and I think I got most/all of those

Reply:
We add prefix FI_ to the codes. So FI_ENODATA is what we want here. Thanks